packages/cli/test/me.endpoints.test.ts

```diff
@@ -0,0 +1,531 @@
+			const shell = await Db.collections.User!.findOneOrFail();
+			const shellAgent = await utils.createAgent(app, shell);
+
+			for (const validPayload of VALID_PATCH_ME_PAYLOADS) {
+				const response = await shellAgent.patch('/me').send(validPayload);
+
+				expect(response.statusCode).toBe(200);
+
+				const {
+					id,
+					email,
+					firstName,
+					lastName,
+					personalizationAnswers,
+					globalRole,
+					password,
+					resetPasswordToken,
+				} = response.body.data;
+
+				expect(validator.isUUID(id)).toBe(true);
+				expect(email).toBe(validPayload.email);
+				expect(firstName).toBe(validPayload.firstName);
+				expect(lastName).toBe(validPayload.lastName);
+				expect(personalizationAnswers).toBeNull();
+				expect(password).toBeUndefined();
+				expect(resetPasswordToken).toBeUndefined();
+				expect(globalRole.name).toBe('owner');
+				expect(globalRole.scope).toBe('global');
+			}
+		});
+
+		test('PATCH /me should fail with invalid inputs', async () => {
+			const shell = await Db.collections.User!.findOneOrFail();
+			const shellAgent = await utils.createAgent(app, shell);
+
+			for (const invalidPayload of INVALID_PATCH_ME_PAYLOADS) {
+				const response = await shellAgent.patch('/me').send(invalidPayload);
+				expect(response.statusCode).toBe(400);
+			}
+		});
+
+		test('PATCH /me/password should succeed with valid inputs', async () => {
+			const shell = await Db.collections.User!.findOneOrFail();
+			const shellAgent = await utils.createAgent(app, shell);
+
+			const validPayloads = Array.from({ length: 3 }, () => ({
+				password: utils.randomValidPassword(),
+			}));
+
+			for (const validPayload of validPayloads) {
+				const response = await shellAgent.patch('/me/password').send(validPayload);
+				expect(response.statusCode).toBe(200);
+				expect(response.body).toEqual(SUCCESS_RESPONSE_BODY);
+			}
+		});
+
+		test('PATCH /me/password should fail with invalid inputs', async () => {
+			const shell = await Db.collections.User!.findOneOrFail();
+			const shellAgent = await utils.createAgent(app, shell);
+
+			const invalidPayloads: Array<{ password?: string }> = Array.from({ length: 3 }, () => ({
+				password: utils.randomInvalidPassword(),
+			}));
+
+			invalidPayloads.push({});
+
+			for (const invalidPayload of invalidPayloads) {
+				const response = await shellAgent.patch('/me/password').send(invalidPayload);
+				expect(response.statusCode).toBe(400);
+			}
+		});
+
+		test('POST /me/survey should succeed with valid inputs', async () => {
+			const shell = await Db.collections.User!.findOneOrFail();
+			const shellAgent = await utils.createAgent(app, shell);
+
+			const validPayloads = [SURVEY, {}];
+
+			for (const validPayload of validPayloads) {
+				const response = await shellAgent.post('/me/survey').send(validPayload);
+				expect(response.statusCode).toBe(200);
+				expect(response.body).toEqual(SUCCESS_RESPONSE_BODY);
+			}
+		});
+	});
+
+	describe('Owner requests', () => {
```

Comment:
what is the difference of an `owner` and a `member` in the `/me` namespace?

Reply:
Specifically, the role expectation differs. But more generally, I view shell, owner and member as separate requests because the role dictates so much of may happen from now on, and that will be even more so the case as we keep adding more role-dependent logic. Let me know if you'd like them merged (branching for role expectation), but it gives me peace of mind that we have fully isolated member tests for all endpoints.